docs/adr/2023-08-18_025-resource-based-api.md

```diff
@@ -0,0 +1,145 @@
+---
+slug: 25
+title: |
+  25. Event-sourced, resource-based API
+authors: []
+tags: [Draft]
+---
+
+## Status
+
+Draft
+
+## Context
+
+- [ADR-3](/adr/3) concluded that a full-duplex communication channels are
+  desirable to interact with a _reactive_ system.
+
+- The Client API communicates several types of messages to clients. Currently
+  this ranges from node-level `PeerConnected`, over head-specific `HeadIsOpen`
+  to messages about transactions like `TxValid`. These messages are all of type
+  `StateChanged`.
+
+- Current capabilities of the API:
+
+  - Clients can retrieve the whole history of `StateChanged` messages or
+    opt-out using a query parameter - all or nothing.
+
+  - There is a welcome message called `Greetings` which is always sent, that
+    contains the last `headStatus`.
+
+  - There exists a `GetUTxO` query-like `ClientInput`, which will respond with a
+    `GetUTxOResponse` containing the confirmed UTxO set in an open head, or (!)
+    the currently committed UTxO set when the head is initializing.
+
+  - While overall `json` encoded, clients can choose choose between `json` or
+    binary (`cbor`) output of `transaction` fields in several of these using a
+    query parameter.
+
+- Many of these features have been added in a "quick and dirty" way, by monkey
+  patching the encoded JSON.
+
+- The current capabalities even do not satisfy all user needs:
+
+  - Need to wade through lots of events to know the latest state (except the
+    very basic `headStatus` from the `Greetings`).
+
+  - Need to poll `GetUTxO` _or_ aggregate confirmed transactions on client side
+    to know the latest UTxO set for constructing transactions.
+
+  - Inclusion of the whole UTxO set in the head is not always desirable and
+    filtering by address would be beneficial. (not addressed in this ADR though)
```

Comment:
Perhaps just mention adding querying capabilities in general and point to the relevant discussion (https://github.com/input-output-hk/hydra/discussions/797)?